dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -588,6 +596,66 @@ public Token addBitbucketToken(@ApiParam(hidden = true) @Auth User user, @QueryP
         }
     }
 
+
+
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/zenodo.org")
+    @ApiOperation(value = "Add a new zenodo.org token, used by accounts page.", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "This is used as part of the OAuth 2 web flow. "
+            + "Once a user has approved permissions for Collaboratory"
+            + "Their browser will load the redirect URI which should resolve here", response = Token.class)
+    public Token addZenodoToken(@ApiParam(hidden = true) @Auth User user, @QueryParam("code") String code) {
+        if (code.isEmpty()) {
+            throw new CustomWebApplicationException("Please provide a Zenodo access code", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        final AuthorizationCodeFlow flow = new AuthorizationCodeFlow.Builder(BearerToken.authorizationHeaderAccessMethod(), HTTP_TRANSPORT,
+                JSON_FACTORY, new GenericUrl(zenodoUrl + "/oauth/token"),
+                new ClientParametersAuthentication(zenodoClientID, zenodoClientSecret), zenodoClientID,
+                zenodoUrl + "/oauth/authorize").build();
```

Comment:
`zenodoUrl + "/oauth/authorize"` is the same thing as zenodoAuthUrl, isn't it?

Reply:
Yes there is overlap that should be cleaned up; I was following an existing pattern.

Reply:
added zenodoAuthUrl to simplify web service zenodo endpoint